### ISSUE TITLE
Specify custom sort algorithm

### ISSUE THREAD

**Issue by MasterEric** [Gudahtt/prettier-plugin-sort-json#27]
Currently the plugin only allows sorting JSON keys using the default sort order. (The default sort order is ascending, built upon converting the elements into strings, then comparing their sequences of UTF-16 code units values.) Ideally, this plugin would allow for the customization of the sorting algorithm used via configuration. 

We could have a set of presets available, but instead for maximum flexibility I would like the user to be able to specify a function with the signature `function(a: any, b: any): int` [as described here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) which will be passed as an argument to `Object.keys(object).sort()` here:

https://github.com/Gudahtt/prettier-plugin-sort-json/blob/main/index.ts#L13

Note that ([as specified by Prettier](https://github.com/prettier/prettier/blob/12bddae20c0bd6c31c53ec44656868f21ec5aca7/src/main/core-options.js#L17)) options can only be an `int`, `boolean`, `choice`, or `path`.

Here is my proposal:

```yaml
printWidth: 90
tabWidth: 4
overrides:
  - files: "**/data/*.json"
    options:
      plugins:
        - "./node_modules/prettier-plugin-sort-json"
      jsonRecursiveSort: true
      jsonObjectSortAlgorithm: "./config/sortFunction.js"
      jsonArraySortAlgorithm: "./config/sortFunctionB.js"
```

`./config/sortFunction.js` is a JavaScript file with contents like such:

```
export default (a, b) => {
  if (a < b) {
    return -1;
  } else if (a > b) {
    return 1;
  }
  return 0;
};
```

This scheme is highly flexible; any desired sorting algorithm can be used, a custom pre-defined order can be defined simply by comparing with a constant array, and with the Prettier configuration system that already exists, different sorting algorithms could be used for different files.

**Comment by MasterEric** [comment]
I am in the process of developing a pull request for this, and found the following caveat.

In the rare circumstance that a user has an object with a key whose value is a number (such as `{ 0: true}`), those numbers will ALWAYS be displayed in ascending order and ALWAYS display before string keys.

[This key order is defined by JavaScript](https://www.stefanjudis.com/today-i-learned/property-order-is-predictable-in-javascript-objects-since-es2015/), and the only way to fix it would be to override the `JSON.stringify` implementation.

Since I did not want to perform a massive overhaul for a rare edge case, I have tried to document this caveat where relevant.

**Comment by MasterEric** [comment]
Resolved by PR #30.

**Comment by MasterEric** [comment]
> This is an interesting workaround, though I am a bit put off by the dynamic require. We don't have any way to know what the imported module might do

My stance is that there aren't any circumstances where this would be a problem. I see the module file as basically an extension of the workspace's prettier configuration, and it's on the user to ensure that their setup makes sense. I don't see how this opens up any malicious code to be run that couldn't just happen by inserting code directly into the project.

**Comment by Gudahtt** [comment]
I have just developed a solution using a JSON configuration file. You can see it here: https://github.com/Gudahtt/prettier-plugin-sort-json/pull/92

>In the rare circumstance that a user has an object with a key whose value is a number (such as { 0: true}), those numbers will ALWAYS be displayed in ascending order and ALWAYS display before string keys.

This should no longer be a problem since I have switched to sorting the AST in #100. I'm no longer using `JSON.stringify`, which is what was imposing this order.

**Comment by Gudahtt** [comment]
This was implemented as part of #92. Let me know what you think!
